ivy/functional/frontends/numpy/statistics/averages_and_variances.py

```diff
@@ -10,6 +10,70 @@
     handle_numpy_out,
 )
 
+# global
+import numpy as np
+from typing import Optional, Sequence, Union
+import ivy
+from ivy.func_wrapper import with_unsupported_dtypes
+from ivy.functional.frontends.numpy.func_wrapper import (
+    to_ivy_arrays_and_back,
+    handle_numpy_dtype,
+    from_zero_dim_arrays_to_scalar,
+    handle_numpy_out,
+)
+
+def nanmedian(
+    a,
+    /,
+     *,
+    axis=None,
+    keepdims=False,
+    out=None,
+    overwrite_input=False
+):
+    is_nan = ivy.isnan(a)
+    axis = tuple(axis) if isinstance(axis, list) else axis
+
+    if not ivy.any(is_nan):
+        ret = ivy.median(a, axis=axis, keepdims=keepdims, out=out, overwrite_input=overwrite_input)
+    else:
+        a = [i for i in a if ivy.isnan(i) is False]
+        ret = ivy.median(a, axis=axis, keepdims=keepdims, out=out, overwrite_input=overwrite_input)
+
+    return ret
+# def nanmedian(
```

Comment:
you should remove these unused code to avoid confusion and extra formatting errors

Reply:
Done